contributors/design-proposals/live-and-inplace-vertical-scaling.md

```diff
@@ -0,0 +1,325 @@
+        cpu: 100m
+        memory: 1Gi
+    limits:
+        cpu: 1000m
+        memory: 1Gi
+    resizePolicy:
+        cpu: LiveResizeable
+        memory: RestartOnly
+```
+
+For the above example, if there is a change to cpu request or limit it can be vertically scaled only if the memory request and limit remained the same, otherwise the RestartOnly policy for memory would override the policy for CPU, and the Pod (container, if container-alone restart is allowed) would need to be restarted.
+
+An example of change in CPU resource `request` size and usage of `resizeAction` annotation with the updated spec.
+
+// before the change
+```yaml
+resources:
+    requests:
+        cpu: **100m**
+        memory: 1Gi
+    limits:
+        cpu: 1000m
+        memory: 1Gi
+    resizePolicy:
+        cpu: LiveResizeable
+        memory: RestartOnly
+```
+
+// after the change
+```yaml
+annotation:
+resizeAction: LiveResizePreferred
+...
+resources:
+    requests:
+        cpu: **400m**
+        memory: 1Gi
+    limits:
+        cpu: 1000m
+        memory: 1Gi
+    resizePolicy:
+        cpu: LiveResizeable
+        memory: RestartOnly
+```
+
+## Combining Stateful Set Update options with Vertical Scaling
+
+`StatefulSet` supports two update options with `spec.updateStrategy.type`, where OnDelete applies the changed spec when restarting a pod after an explicit delete command and the RollingUpdate is applied by the controller by restarting each of the pods that are members of the Stateful set with the changed values, in reverse ordinal sequence (see [https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#updating-statefulsets](https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#updating-statefulsets) for more details).
+
+In the table below we propose how the resource resizing directives for vertical scaling can be applied in conjunction with the 'StatefulSet' update strategy.
+
+|resizePolicy (resizeAction)|OnDelete|RollingUpdate|
+|---|---|---|
+|RestartOnly (or Action=Restart)|Resize resource with restart (with delete command)|Resize resource with restart (with allowed spec update commands)|
+|LiveResizeable (and Action=LiveResize)|Resize resource live only (with allowed spec update commands)|Resize resource live only (with allowed spec update commands)|
+LiveResizeable (and Action=LiveResizePreferred or no Action specified)|Resize resource with live resize preferred (with allowed spec update commands), i.e., resize resource with restart (with delete command) if not able to resize live.|Resize resource with live resize preferred (with allowed spec update commands), i.e., resize resource with restart (managed by controller) if not able to resize live.|
+
+## Desired Approach
+
+Any valid method to update the pod spec should be applicable for vertical scaling, e.g., using kubectl commands set, patch, apply, edit.
+Logs associated with the pod will capture the failure/success of the resize command.
```

Comment:
I am not sure I follow this.  Can you elaborate?

Reply:
I am also confused about this, use logs to capture the resize command result? any consider of `PodCondition` or `Events`?

Reply:
Basically, we see resizing a pod as a just another pod spec update. So, such resizing is requested by the existing spec update methods like kubectl patch or apply, which are currently used to change statefulset specs.

What we mean by 'Logs' is general logs/messages generated by k8s components (like apiserver, kubelet,  scheduler, and so on) and that we check with kubectl describe. And yes, the status of resizing can be also checked with a new PodCondition, called PodResized.